### DISCUSSION TITLE
Cross module cast

### DISCUSSION THREAD

**Discussion by luxuncang** [microsoft/pyright#3771] [General]
Tips for re entering object class type when `import` packages

* graia.py

```python
class Graia:
    ...
```

* patch.py

```python
import typing
import graia

class Patch:
   ...
#graia.Graia = Patch
typing.Graia = typing.cast(typing.Type[Patch], graia.Graia)
```

* main.py
```python
import patch
patch.patch()
import graia

graia.Graia # Prompt graia.Graia instead of patch.Patch
```

I don't know much about Python type system, and I feel confused in cross module cast

**Comment by erictraut** [comment]
Moving this to "discussion" since it's not a bug report or feature request.

I'm not sure what you're trying to do here. It looks like you are trying to monkey patch a module and overwrite one of its symbols? If so, here's how you would do this.

```python
cast(graia, Any).Graia = Patch
```
